main.js

```diff
@@ -0,0 +1,133 @@
+function Book(title, author, pages) {
+  this.title = title;
+  this.author = author;
+  this.pages = pages;
+  this.is_read = false;
+}
+
+Book.prototype.toggleStatus = function toggleStatus() {
+  this.is_read = !this.is_read;
+};
+
+function getBookList() {
+  const items = JSON.parse(localStorage.getItem('bookList'));
+  const parsedItems = [];
+  items.forEach(element => {
+    const book = new Book(element.title, element.author, element.pages);
+    book.is_read = element.is_read;
+    parsedItems.push(book);
+  });
+  return parsedItems;
+}
+
+function updateBookList(array) {
+  localStorage.setItem('bookList', JSON.stringify(array));
+}
+
+function changeStatus(e) {
+  const index = e.path[3].getAttribute('data-attribute');
```

Comment:
e here should be the event on the DOM object, which is the button in this case, I do not think there is a path property, anyway this throws an error